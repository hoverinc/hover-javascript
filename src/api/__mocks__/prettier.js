module.exports = {
  format: jest.fn(() => '[formatted-code]'),
}
