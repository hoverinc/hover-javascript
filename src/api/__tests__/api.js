import cases from 'jest-in-case'
import prettier from 'prettier'
import {format} from '../'

cases(
  'format',
  ({source = 'const id = x => x', options = {}}) => {
    // beforeEach
    format(source, options)

    try {
      expect(prettier.format).toHaveBeenCalledTimes(1)
      expect(prettier.format).toHaveBeenCalledWith(source, expect.any(Object))

      const [call] = prettier.format.mock.calls

      expect(call[1]).toMatchSnapshot()
    } catch (error) {
      throw error
    } finally {
      // afterEach

      prettier.format.mockReset()
      jest.resetModules()
    }
  },
  {
    'calls `prettier.format` with internal Prettier configuration': {},
    'merges provided `options` with internal Prettier configuration': {
      /** @type {import('prettier').Options} */
      options: {printWidth: 100, arrowParens: 'always', parser: 'css'},
    },
  },
)
