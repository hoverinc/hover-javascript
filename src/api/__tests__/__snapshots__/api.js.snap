// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format calls \`prettier.format\` with internal Prettier configuration 1`] = `
Object {
  "arrowParens": "avoid",
  "bracketSameLine": false,
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxSingleQuote": false,
  "printWidth": 80,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all",
  "useTabs": false,
}
`;

exports[`format merges provided \`options\` with internal Prettier configuration 1`] = `
Object {
  "arrowParens": "always",
  "bracketSameLine": false,
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxSingleQuote": false,
  "parser": "css",
  "printWidth": 100,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all",
  "useTabs": false,
}
`;
